osfoffline/sync/remote.py

```diff
@@ -107,14 +111,25 @@ def _orphan_children(self, node, remote_children):
         locally for which the remote Node has been deleted are explicitly removed
         from OSFO's auditing and will be ignored.
         """
-        children_ids = map(lambda c: c.id, remote_children)
+        children_ids = [c.id for c in remote_children]
         for record in node.children:
             if record.id not in children_ids:
                 Session().delete(record)
+            else:
+                remote_child = OSFClient().get_node(record.id)
+                self._orphan_children(record, remote_child.get_children(lazy=False))
 
     def _preprocess_node(self, node, *, delete=True):
         nodes = [node]
-        remote_node = OSFClient().get_node(node.id)
+        try:
+            remote_node = OSFClient().get_node(node.id)
+        except ClientLoadError as err:
+            if err.status in (http.client.NOT_FOUND, http.client.GONE):
+                # cascade should automagically delete children
+                Session().delete(node)
```

Comment:
It appears that calling delete doesn't actually delete from the underlying DB. You may need to use `Session().commit` or `flush` after this to avoid error messages and various other odd behavior.